src/Scaffolding/VS.Web.CG.Mvc/Identity/IdentityHelper.cs

```diff
@@ -0,0 +1,154 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.DotNet.Scaffolding.Shared.Project;
+
+namespace Microsoft.VisualStudio.Web.CodeGenerators.Mvc.Identity
+{
+    /// <summary>
+    /// Identity helper with static methods to help IdentityGenerator and BlazorIdentityGenerator with scaffoding.
+    /// </summary>
+    internal static class IdentityHelper
+    {
+        internal static bool IsTypeDerivedFromIdentityUser(Type type)
+        {
+            var parentType = type.BaseType;
+            while (parentType != null && parentType != typeof(object))
+            {
+                if (parentType.FullName == "Microsoft.AspNetCore.Identity.IdentityUser"
+                    && parentType.Assembly.GetName().Name == "Microsoft.Extensions.Identity.Stores")
+                {
+                    return true;
+                }
+
+                parentType = parentType.BaseType;
+            }
+
+            return false;
+        }
+
+        internal static string GetNamespaceFromTypeName(string dbContext)
+        {
+            if (dbContext.LastIndexOf('.') == -1)
+            {
+                return null;
+            }
+
+            return dbContext.Substring(0, dbContext.LastIndexOf('.'));
+        }
+
+        internal static string GetClassNameFromTypeName(string dbContext)
+        {
+            return dbContext.Split(new char[] { '.' }, StringSplitOptions.RemoveEmptyEntries).Last();
```

Comment:
I'd generally suggest not doing a Split if you don't care about all of the parts. If you're just going after the last one, you could do effectively the opposite of what you did above in `GetNamespaceFromTypeName` above. And then the two methods next to each other would be more consistent as well.

Reply:
sounds good, this file also has no unit tests so adding those.